File: CWM.Model/Models/Appointment.cs
using CWM.Core.Models.Enums;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json;
using System.Threading.Tasks;

namespace CWM.Core.Models
{
    public class Appointment : Base
    {
        public string Description { get; set; } = string.Empty;
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public User? User { get; set; } = new();
        public AppointmentType? AppointmentType { get; set; } = new();
        public Vehicle? Vehicle { get; set; } = new();


        public string ToJson()
        => JsonSerializer.Serialize(this, new JsonSerializerOptions() { PropertyNamingPolicy = JsonNamingPolicy.CamelCase });
    }
}

File: CWM.UI/cwm_desktop_mobile/lib/models/appointment.dart
import 'package:cwm_desktop_mobile/models/appointment_type.dart';
import 'package:cwm_desktop_mobile/models/user.dart';
import 'package:cwm_desktop_mobile/models/vehicle.dart';
import 'package:json_annotation/json_annotation.dart';

part 'appointment.g.dart';

@JsonSerializable()
class Appointment {
  int id;
  String description;
  DateTime startDate;
  DateTime endDate;
  AppointmentType? appointmentType;
  User? user;
  Vehicle? vehicle;

  Appointment(this.id, this.description, this.startDate, this.endDate,
      this.appointmentType);

  factory Appointment.fromJson(Map<String, dynamic> json) =>
      _$AppointmentFromJson(json);

  Map<String, dynamic> toJson() => _$AppointmentToJson(this);
}
